pkg/model/manifest_test.go

```diff
@@ -404,3 +407,206 @@ func TestSetManifestDefaultsFromDev(t *testing.T) {
+				},
+			},
+			expectedManifest: ManifestBuild{
+				"test4": &BuildInfo{
+					Context:    ".",
+					Dockerfile: "test4.Dockerfile",
+				},
+			},
+			dockerfileCreated: filepath.Join(mockDir, "test4.Dockerfile"),
+			expectedError:     true,
+		},
+		{
+			name: "Manifest: Dockerfile empty / Paths: Dockerfile relative to context",
+			currentManifest: ManifestBuild{
+				"test5": &BuildInfo{
+					Context: buildName,
+				},
+			},
+			expectedManifest: ManifestBuild{
+				"test5": &BuildInfo{
+					Context:    buildName,
+					Dockerfile: filepath.Join(buildName, "Dockerfile"),
+				},
+			},
+			dockerfileCreated: filepath.Join(buildName, "Dockerfile"),
+			expectedError:     false,
+		},
+		{
+			name: "Manifest: Dockerfile empty / Paths: Dockerfile NOT relative to context",
+			currentManifest: ManifestBuild{
+				"test6": &BuildInfo{
+					Context: buildName,
+				},
+			},
+			expectedManifest: ManifestBuild{
+				"test6": &BuildInfo{
+					Context:    buildName,
+					Dockerfile: "Dockerfile",
+				},
+			},
+			dockerfileCreated: filepath.Join(mockDir, "test6.Dockerfile"),
+			expectedError:     true,
+		},
+		{
+			name: "Manifest: Context and Dockerfile filled / Paths: Dockerfile relative to context",
+			currentManifest: ManifestBuild{
+				"test7": &BuildInfo{
+					Context:    buildName,
+					Dockerfile: "test7.Dockerfile",
+				},
+			},
+			expectedManifest: ManifestBuild{
+				"test7": &BuildInfo{
+					Context:    buildName,
+					Dockerfile: filepath.Join(buildName, "test7.Dockerfile"),
+				},
+			},
+			dockerfileCreated: filepath.Join(buildName, "test7.Dockerfile"),
+			expectedError:     false,
+		},
+		{
+			name: "Manifest: Context and Dockerfile filled / Paths: Dockerfile NOT relative to context",
+			currentManifest: ManifestBuild{
+				"test8": &BuildInfo{
+					Context:    buildName,
+					Dockerfile: "test8.Dockerfile",
+				},
+			},
+			expectedManifest: ManifestBuild{
+				"test8": &BuildInfo{
+					Context:    buildName,
+					Dockerfile: "test8.Dockerfile",
+				},
+			},
+			dockerfileCreated: "test8.Dockerfile",
+			expectedError:     true,
+		},
+	}
+
+	for _, tt := range tests {
+
+		t.Run(tt.name, func(t *testing.T) {
+
+			if tt.dockerfileCreated != "" {
+				defer removeFile(tt.dockerfileCreated)
+				// create test dockerfile
+				_, err := os.Create(tt.dockerfileCreated)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				log.Printf("created docker file: %s", tt.dockerfileCreated)
+			}
+
+			buildKey := reflect.ValueOf(tt.currentManifest).MapKeys()[0].String()
+			errManifest := tt.currentManifest[buildKey].setBuildDefaults()
+
+			if errManifest != nil {
+				oktetoLog.Warning(fmt.Sprintf("Build '%s': %s", buildKey, errManifest.Error()))
+			}
+
+			if errManifest != nil && !tt.expectedError {
+				t.Fatal(err)
+			}
+
+			if errManifest == nil && tt.expectedError {
+				t.Fatal("error not thrown")
+			}
+
+			assert.Equal(t, tt.expectedManifest, tt.currentManifest)
```

Comment:
```suggestion
			assert.DeepEqual(t, tt.expectedManifest, tt.currentManifest)
```

Reply:
![image](https://user-images.githubusercontent.com/16018223/170506674-d21767d9-7bc3-4a17-a5d3-09c13e63e353.png)

Reply:
Sorry, it was `reflect.DeepEqual`

Reply:
`assert.Equal` uses `reflect.DeepEqual` under the hood and provides a better error formatting to the dev. I think we should keep going with `assert.Equal`

Reply:
Yes, I tried deep equal but the output wasn't that clear...